From: Jiri Kosina <jkosina@suse.cz>
Subject: [PATCH] kabi: ignore struct bpf_insn_aux_data change
Patch-mainline: No, SUSE-specific
References: bsc#1087082

Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
Replace `int sanitize_stack_off` with `bool sanitize_stack_spill` to adapte to
changes in introduced in "bpf: Fix leakage due to insufficient speculative
store bypass mitigation"

Signed-off-by: Shung-Hsi Yu <shung-hsi.yu@suse.com>
---
 include/linux/bpf_verifier.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/bpf_verifier.h
+++ b/include/linux/bpf_verifier.h
@@ -130,8 +130,10 @@ struct bpf_insn_aux_data {
 	};
 	int ctx_field_size; /* the ctx field size for load insn, maybe 0 */
 	bool seen; /* this insn was processed by the verifier */
-	bool sanitize_stack_spill; /* subject to Spectre v4 sanitation */
 	u8 alu_state; /* used in combination with alu_limit */
+#ifndef __GENKSYMS__
+	bool sanitize_stack_spill; /* subject to Spectre v4 sanitation */
+#endif
 };
 
 #define MAX_USED_MAPS 64 /* max number of maps accessed by one eBPF program */
